watergun_assassin/user_profile/views.py

```diff
@@ -0,0 +1,18 @@
+from django.shortcuts import render
+from django.http import JsonResponse
+from .models import UserProfile
+from django.views.decorators.csrf import csrf_exempt
```

Comment:
This is an extreme nitpick but could we move this last import up above importing the UserProfile? I think it will make it look neater, plus I generally prefer having all package imports in one block and then imports from local files after all those.